src/envoy/http/mixer/filter.cc

```diff
@@ -245,6 +245,7 @@ void Filter::completeCheck(const CheckResponseInfo& info) {
   if (nullptr != headers_) {
     UpdateHeaders(*headers_, route_directive_.request_header_operations());
     headers_ = nullptr;
+    decoder_callbacks_->clearRouteCache();
```

Comment:
clear only `request_header_operations` is not empty?

Reply:
Done. Forgot to add this check.